libs/expo/betterangels/src/lib/screens/AddEditClient/PersonalInfo/index.tsx

```diff
@@ -17,6 +21,49 @@ interface IPersonalInfoProps {
 
 export default function PersonalInfo(props: IPersonalInfoProps) {
   const { scrollRef, expanded, setExpanded } = props;
+  const { id: clientProfileId } = useLocalSearchParams();
+  const {
+    formState: { errors },
+    watch,
+    setError,
+    clearErrors,
+  } = useFormContext();
+  const [checkForDuplicates, setCheckForDuplicates] = useState(false);
+  const californiaId = watch('californiaId');
+  const californiaIdLength = 8;
+  const { data } = useClientProfilesQuery({
+    skip: !checkForDuplicates,
+    variables: {
+      filters: {
+        searchClient: {
+          excludedClientProfileId: (clientProfileId as string) || null,
+          californiaId: californiaId,
+        },
+      },
+    },
+    fetchPolicy: 'network-only',
+    nextFetchPolicy: 'network-only',
+  });
+
+  useEffect(() => {
+    const shouldCheck =
+      californiaId &&
+      californiaId.length === californiaIdLength &&
+      !errors['californiaId'];
+
+    setCheckForDuplicates(shouldCheck);
+  }, [californiaId, errors]);
+
+  useEffect(() => {
+    if (data && data?.clientProfiles?.length > 0) {
```

Comment:
alternatively, we could check if the current client == the returned client. Then we could remove the excluded id param. I think it's faster to do this work on the backend, so I went with that approach